### DISCUSSION TITLE
Find what files changed in a pushed commit?

### DISCUSSION THREAD

**Discussion by frigid-dev** [community/community#25170] [Actions]
<p>I want to get the file names or relative path of the changed files triggered by push. I am already filtering the action with the on.push.paths functionality in the workflow. I need to find what specific files changed as I am going to run a process on them and would like to only run it on the changed files.</p>
<p>  </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Please use below command to get the changed files in the push commit, please refer to this <a href="https://stackoverflow.com/questions/424071/how-to-list-all-the-files-in-a-commit" rel="nofollow noopener">link</a> for more details.</p>
<pre><code>git diff-tree --no-commit-id --name-only -r ${{ github.sha }}
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181089225-0859ec55-669e-4916-900f-3a271f6b76f1.png" alt="changedfiles.png" data-base62-sha1="U9diJOAA3xyfRbVJxaouXSuu8Z" width="629" height="219"></p>

**Comment by frigid-dev** [comment]
<p>Thanks that worked great. Now I just need to figure out how to pipe that into the next step. </p>

**Comment by tigelane** [comment]
<p>"Thanks that worked great. Now I just need to figure out how to pipe that into the next step. "</p>
<p>I’m working on the same thing.  Please post if you find a solution and I’ll do the same.</p>

**Comment by andyli** [comment]
<p>What if I pushed more than one commits at once? What would ${{ github.sha }} contain?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/andyli">@andyli</a> ,</p>
<p>For multi commits, you can find my anwser in this <a href="https://github.community/t/check-pushed-file-changes-with-git-diff-tree-in-github-actions/17220">ticket</a>.</p>
<p>Thanks.</p>

**Comment by mghazizadeh** [comment]
<p>Monitor github file changes on commits, PRs, etc (on any repo) and get a notification using webhooks, look for patterns, review diffs, take actions, etc. </p>
<p>Source code is available here: <a href="https://github.com/DevScoreInc/samples/tree/master/github-file-monitor" rel="nofollow noopener">https://github.com/DevScoreInc/samples/tree/master/github-file-monitor</a></p>
<p>Demo is here: <a href="https://youtu.be/6HgxIkT8EQ4" rel="nofollow noopener">https://youtu.be/6HgxIkT8EQ4</a></p>
<p>Free to deploy and use. </p>
